pkg/gofr/http_server_test.go

```diff
@@ -85,3 +85,76 @@ func TestRegisterProfillingRoutes(t *testing.T) {
 		assert.Equal(t, http.StatusOK, rr.Code)
 	}
 }
+
+func TestShutdown_ServerStopsListening(t *testing.T) {
+	// Create a mock router and add a new route
+	router := &gofrHTTP.Router{}
+	router.Add(http.MethodGet, "/", http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+
+	// Create a mock container
+	c := &container.Container{
+		Logger: logging.NewLogger(logging.INFO),
+	}
+
+	// Create an instance of httpServer
+	server := &httpServer{
+		router: router,
+		port:   8080,
+	}
+
+	// Start the server
+	go server.Run(c)
+
+	// Create a context with a timeout to test the shutdown
+	ctx, cancel := context.WithTimeout(context.Background(), 200*time.Millisecond)
+	defer cancel()
+
+	errChan := make(chan error, 1)
+	go func() {
+		time.Sleep(100 * time.Millisecond)
+		errChan <- server.Shutdown(ctx)
+	}()
+
+	err := <-errChan
+
+	assert.NoError(t, err, "TEST Failed.\n")
+}
+
+func TestShutdown_ServerContextDeadline(t *testing.T) {
+	// Create a mock router and add a new route
+	router := &gofrHTTP.Router{}
+	router.Add(http.MethodGet, "/", http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+
+	// Create a mock container
+	c := &container.Container{
+		Logger: logging.NewLogger(logging.INFO),
+	}
+
+	// Create an instance of httpServer
+	server := &httpServer{
+		router: router,
+		port:   8080,
+	}
+
+	// Start the server
+	go server.Run(c)
+
+	// Create a context with a timeout to test the shutdown
+	ctx, cancel := context.WithTimeout(context.Background(), 200*time.Millisecond)
```

Comment:
Can we reduce the timeouts in tests? I don't think even 200 ms is required.

Reply:
done reduced